### PR TITLE
Jukebox new array style

### PR DESCRIPTION
**Pull request by Brycey92**
### Test Instructions

Spammed buttons on both emulator and hardware

### Readiness Checklist
- [x] I have run `make format` to format the changes
- [x] I have compiled the firmware and the changes have no warnings
- [x] I have compiled the emulator and the changes have no warnings
- [x] I have run `make cppcheck` and checked that `cppcheck_result.txt` has no warnings for the changes
- [x] I have added doxygen comments to any code used by more than one Swadge mode. This includes `/*! \file` comments with Design Philosophy, Usage, and Example sections for new headers.
- [x] I have run `make docs` and checked that `doxy_warnings.txt` has no warnings for the new code

### CODE REVIEW

File: main/modes/jukebox/jukebox.h [hunk 1]
```diff
@@ -12,4 +12,6 @@
 
 extern swadgeMode_t jukeboxMode;
 
+extern const char jukeboxName[];
```
Comment (AEFeinstein) [inline main/modes/jukebox/jukebox.h:15]: Love the consistency, but really?

File: main/modes/jukebox/jukebox.c [hunk 2]
```diff
@@ -150,694 +145,490 @@ static const char str_play[]       = ": Play";
...
-    jukebox->musicCategories[catIdx].songs = calloc(jukebox->musicCategories[catIdx].numSongs, sizeof(jukeboxSong));
-
-    // Credits songs
-
-    jukebox->musicCategories[catIdx].songs[songIdx].name = "Credits";
-    loadSong("credits.sng", &jukebox->musicCategories[catIdx].songs[songIdx].song, true);
-    jukebox->musicCategories[catIdx].songs[songIdx].song.shouldLoop = false;
-    songIdx++;
+        .filename = "stereo.sng",
+        .name     = "DELETE ME",
```
Comment (AEFeinstein) [inline main/modes/jukebox/jukebox.c:151]: Um, what's this?

**Review by AEFeinstein — APPROVED** [review]

File: main/modes/jukebox/jukebox.c (hunk 2 above)
Comment (Brycey92) [reply main/modes/jukebox/jukebox.c:151]: Needed something in there or it would crash 😎

File: main/modes/jukebox/jukebox.h (hunk 1 above)
Comment (Brycey92) [reply main/modes/jukebox/jukebox.h:15]: I'm pretty sure that was an artifact of me working on it at 2am lol

**Pull request merged**